# Study 952 — After-Tax Equivalent — the teardown

The linear-in-tau identity and the closed-form break-even, the five-bracket ladder, HAC *t* and block-bootstrap CIs, the era cut, the state / capital-gains / cost / borrow sweeps, the single-lag overlay, and the live synthetic control. Every real-tape number is frozen from `docs/results.md` (Fingerprint `0f379866a58d`), monthly 2004-02 → 2026-06. Every tax rate is a labelled PROXY.

In [1]:
R = {'start': '2004-02', 'end': '2026-06', 'n_months': 269, 'fp': '0f379866a58d', 'inc_mub': 2.77, 'inc_vteb': 2.32, 'inc_sub': 1.27, 'inc_hyd': 5.43, 'inc_agg': 3.2, 'inc_lqd': 4.07, 'inc_vcit': 3.63, 'inc_bil': 1.35, 'inc_mub_on_vcit': 2.66, 'inc_agg_on_mub': 3.0, 'inc_lqd_on_mub': 3.9, 'be_mub_vcit': 35.0, 'be_mub_lqd': 29.7, 'be_vteb_vcit': 28.9, 'be_mub_agg': -6.9, 'be_vteb_agg': -18.7, 'be_sub_bil': -24.0, 'be_hyd_lqd': -21.5, 'beci_mub_vcit': (-10.9, 81.5), 'beci_mub_lqd': (-25.9, 79.3), 'beci_vteb_vcit': (-27.3, 82.7), 'beci_mub_agg': (-52.1, 37.3), 'beci_vteb_agg': (-78.1, 38.6), 'beci_sub_bil': (-115.3, 34.6), 'beci_hyd_lqd': (-94.8, 58.1), 'p_above_top_mub_vcit': 0.41, 'p_below_zero_mub_vcit': 0.06, 'price_leg_mub_vcit': -2.53, 'price_leg_mub_vcit_t': -0.36, 'price_leg_mub_agg': 3.64, 'price_leg_mub_agg_t': 0.68, 'ibe_mub_vcit': 26.7, 'ibeci_mub_vcit': (23.4, 29.7), 'ibe_mub_lqd': 28.9, 'ibeci_mub_lqd': (27.1, 30.7), 'ibe_vteb_vcit': 34.3, 'ibeci_vteb_vcit': (31.0, 38.2), 'ibe_mub_agg': 7.6, 'ibeci_mub_agg': (2.7, 12.0), 'dec_mub_agg': (1.73, 0.31, 10.21, 5.72, 11.94, 2.22, 85.5, 0.3), 'dec_vteb_agg': (4.34, 0.65, 9.47, 4.74, 13.81, 2.1, 68.6, 0.37), 'dec_sub_bil': (2.49, 0.69, 4.22, 6.87, 6.71, 2.02, 62.9, 1.87), 'dec_hyd_lqd': (6.77, 0.57, 12.83, 6.01, 19.6, 1.67, 65.5, 0.09), 'dec_mub_vcit': (-10.6, -1.5, 12.35, 6.32, 1.75, 0.25, None, 0.32), 'pre_mub_vcit': -10.6, 'pre_mub_vcit_t': -1.5, 'pre_mub_lqd': -9.65, 'pre_mub_lqd_t': -1.04, 'pre_mub_agg': 1.73, 'pre_mub_agg_t': 0.31, 'n_mub_vcit': 199, 'd_mub_vcit': 1.72, 't_mub_vcit': 0.25, 'ci_mub_vcit_lo': -12.5, 'ci_mub_vcit_hi': 15.6, 'sh_mub_vcit_m': 0.47, 'sh_mub_vcit_t': 0.35, 'ann_mub_vcit': 0.21, 'n_mub_lqd': 225, 'd_mub_lqd': 3.59, 't_mub_lqd': 0.39, 'n_mub_agg': 225, 'd_mub_agg': 11.91, 't_mub_agg': 2.22, 'ann_mub_agg': 1.43, 'ci_mub_agg_lo': 0.9, 'ci_mub_agg_hi': 22.5, 'sh_mub_agg_m': 0.48, 'sh_mub_agg_t': 0.23, 'n_vteb_agg': 130, 'd_vteb_agg': 13.77, 't_vteb_agg': 2.09, 'n_sub_bil': 211, 'd_sub_bil': 6.69, 't_sub_bil': 2.01, 'n_hyd_lqd': 208, 'd_hyd_lqd': 19.57, 't_hyd_lqd': 1.66, 'ladder_rates': [0.0, 24.0, 27.8, 35.8, 40.8], 'ladder_vcit': [-10.63, -3.37, -2.22, 0.2, 1.72], 'ladder_vcit_t': [-1.5, -0.48, -0.32, 0.03, 0.25], 'ladder_agg': [1.7, 7.71, 8.66, 10.66, 11.91], 'ladder_agg_t': [0.31, 1.42, 1.6, 1.98, 2.22], 'era_vcit_e_n': 85, 'era_vcit_e': -2.4, 'era_vcit_e_t': -0.23, 'era_vcit_l_n': 114, 'era_vcit_l': 4.73, 'era_vcit_l_t': 0.51, 'era_agg_e_n': 111, 'era_agg_e': 9.96, 'era_agg_e_t': 1.25, 'era_agg_l_n': 114, 'era_agg_l': 13.76, 'era_agg_l_t': 1.92, 'state_none': 1.72, 'state_ca_out': 2.47, 'state_ca_in': 4.53, 'state_ca_in_t': 0.65, 'cg_0': 1.72, 'cg_15': 2.1, 'cg_238': 2.32, 'floor_on': 1.717, 'floor_off': 1.716, 'cost0': 1.75, 'cost25': 1.5, 'borrow0_vcit': 1.72, 'borrow25_vcit': -0.37, 'borrow100_vcit': -6.62, 'borrow0_agg': 11.91, 'borrow25_agg': 9.83, 'borrow25_agg_t': 1.83, 'borrow100_agg': 3.58, 'ov_n': 199, 'ov_switches': 3, 'ov_in': 93.0, 'ov_ann': 3.33, 'ov_hold': 3.05, 'ov_edge': 2.31, 'ov_t': 0.9, 'syn_planted_true': 33.33, 'syn_planted_got': 38.97, 'syn_planted_maxerr': 10.3, 'syn_planted_pre_t': -8.29, 'syn_null_true': 0.0, 'syn_null_got': 8.05, 'syn_null_maxerr': 9.8, 'syn_null_pre_t': -1.46}

## The accounting, in one line

`after_tax = price x (1 - capgain) + income x (1 - tau_income)`, with the income leg reconstructed as `total_return - price_return` per month. `tau_income` is 0 for munis federally (state only, and only on the out-of-state slice), `fed + NIIT + state` for taxable credit, `fed + NIIT` for T-bills. With the default `capgain = 0` this collapses to `total - tau x income`, so the after-tax difference between a muni and a taxable leg is **exactly affine in tau**:

`d(tau) = d(0) + tau x mean(income_taxable)`

which is why the break-even needs no search: `tau* = -mean(d(0)) / mean(income_taxable)`.

> 💡 **In plain words.** Raise the tax rate and the only thing that changes is how much the taxable bond's coupon is docked. That makes the muni-minus-taxable gap a straight line, and the break-even is just where the line crosses zero.

## Break-even by pairing — with the interval that decides whether to quote it

`tau*` is a **ratio of two sample means**, and its numerator (the pre-tax total-return difference) never clears |*t*| ≥ 2 in any pairing on this tape. The point estimate is therefore far more precise-looking than the data warrants. The block bootstrap below is the correction; `p>40.8%` is the share of resamples in which **no US bracket is high enough**.

In [2]:
rows = [('MUB','VCIT', R['pre_mub_vcit'], R['pre_mub_vcit_t'], R['be_mub_vcit'], R['beci_mub_vcit']),
        ('MUB','LQD ', R['pre_mub_lqd'],  R['pre_mub_lqd_t'],  R['be_mub_lqd'],  R['beci_mub_lqd']),
        ('VTEB','VCIT', None, None, R['be_vteb_vcit'], R['beci_vteb_vcit']),
        ('MUB','AGG ', R['pre_mub_agg'],  R['pre_mub_agg_t'],  R['be_mub_agg'],  R['beci_mub_agg']),
        ('VTEB','AGG ', None, None, R['be_vteb_agg'], R['beci_vteb_agg']),
        ('SUB','BIL ', None, None, R['be_sub_bil'], R['beci_sub_bil']),
        ('HYD','LQD ', None, None, R['be_hyd_lqd'], R['beci_hyd_lqd'])]
for a, b, pre, t, be, ci in rows:
    pre_s = '   n/a        ' if pre is None else 'pre-tax %+6.2f bps (t=%+5.2f)' % (pre, t)
    print('%-5s vs %-5s  %s  tau* %+6.1f%%  95%% CI [%+7.1f%%, %+6.1f%%]'
          % (a, b, pre_s, be, ci[0], ci[1]))
print()
print('MUB/VCIT: %.0f%% of resamples put tau* ABOVE the top US rate (40.8%%),'
      % (100*R['p_above_top_mub_vcit']))
print('          %.0f%% put it BELOW zero. The total-return break-even is not identified.'
      % (100*R['p_below_zero_mub_vcit']))
print('The width is the PRICE legs: MUB-VCIT price diff %+.2f bps/mo (t=%+.2f).'
      % (R['price_leg_mub_vcit'], R['price_leg_mub_vcit_t']))

MUB   vs VCIT   pre-tax -10.60 bps (t=-1.50)  tau*  +35.0%  95% CI [  -10.9%,  +81.5%]
MUB   vs LQD    pre-tax  -9.65 bps (t=-1.04)  tau*  +29.7%  95% CI [  -25.9%,  +79.3%]
VTEB  vs VCIT      n/a          tau*  +28.9%  95% CI [  -27.3%,  +82.7%]
MUB   vs AGG    pre-tax  +1.73 bps (t=+0.31)  tau*   -6.9%  95% CI [  -52.1%,  +37.3%]
VTEB  vs AGG       n/a          tau*  -18.7%  95% CI [  -78.1%,  +38.6%]
SUB   vs BIL       n/a          tau*  -24.0%  95% CI [ -115.3%,  +34.6%]
HYD   vs LQD       n/a          tau*  -21.5%  95% CI [  -94.8%,  +58.1%]

MUB/VCIT: 41% of resamples put tau* ABOVE the top US rate (40.8%),
          6% put it BELOW zero. The total-return break-even is not identified.
The width is the PRICE legs: MUB-VCIT price diff -2.53 bps/mo (t=-0.36).


> ⚠️ **The sweep fallacy.** `tau*` is invariant to the bracket, the state rate, the in-state share and the capital-gains rate — because *no tax parameter enters its numerator*. "Stable under every sweep" is a statement about the arithmetic and carries **zero** information about precision. Only resampling the tape does (`strategy.breakeven_ci`, regression-tested both ways).

## The half the tape does pin down: the income-leg break-even

Drop the price legs and ask `tau* = 1 - y_muni / y_taxable` on realised monthly distributions net of fund fees. Income streams are smooth, so this is tight — and it is the study's one durable number.

In [3]:
for a, b, ibe, ci in [('MUB','VCIT', R['ibe_mub_vcit'], R['ibeci_mub_vcit']),
                      ('MUB','LQD ', R['ibe_mub_lqd'],  R['ibeci_mub_lqd']),
                      ('VTEB','VCIT', R['ibe_vteb_vcit'], R['ibeci_vteb_vcit']),
                      ('MUB','AGG ', R['ibe_mub_agg'],  R['ibeci_mub_agg'])]:
    print('%-5s vs %-5s  income-leg tau* %+6.1f%%   95%% CI [%+5.1f%%, %+5.1f%%]'
          % (a, b, ibe, ci[0], ci[1]))
print()
print('Note MUB/AGG: on income alone AGG OUT-yields MUB (tau* = %+.1f%%, CI strictly'
      % R['ibe_mub_agg'])
print('positive), so the %+.1f%% total-return break-even that made MUB look like a'
      % R['be_mub_agg'])
print('pre-tax winner is produced entirely by a price-leg diff of %+.2f bps/mo (t=%+.2f).'
      % (R['price_leg_mub_agg'], R['price_leg_mub_agg_t']))

MUB   vs VCIT   income-leg tau*  +26.7%   95% CI [+23.4%, +29.7%]
MUB   vs LQD    income-leg tau*  +28.9%   95% CI [+27.1%, +30.7%]
VTEB  vs VCIT   income-leg tau*  +34.3%   95% CI [+31.0%, +38.2%]
MUB   vs AGG    income-leg tau*   +7.6%   95% CI [ +2.7%, +12.0%]

Note MUB/AGG: on income alone AGG OUT-yields MUB (tau* = +7.6%, CI strictly
positive), so the -6.9% total-return break-even that made MUB look like a
pre-tax winner is produced entirely by a price-leg diff of +3.64 bps/mo (t=+0.68).


## Where the *t*-stat comes from — `d(tau) = d(0) + tau x i_taxable`

The second term is a coupon stream: large in the mean, near-constant in time. Adding a near-deterministic constant to a noisy series lifts the mean without lifting the variance, so **HAC *t* climbs with the bracket by construction**.

In [4]:
print('pair        pre-tax (t)      + tax term (sd)     = total (t)      mean%   var%')
for name, d in [('MUB-AGG ', R['dec_mub_agg']), ('VTEB-AGG', R['dec_vteb_agg']),
                ('SUB-BIL ', R['dec_sub_bil']), ('HYD-LQD ', R['dec_hyd_lqd']),
                ('MUB-VCIT', R['dec_mub_vcit'])]:
    print('%s  %+6.2f (%+5.2f)   %+6.2f (sd %4.2f)   %+6.2f (%+5.2f)  %6s  %5.2f'
          % (name, d[0], d[1], d[2], d[3], d[4], d[5],
             '--' if d[6] is None else '%.1f' % d[6], d[7]))
print()
print('Every row crossing |t| = 2 does so on a term carrying < 2% of the variance.')
print('That is a test of whether the tax code is nonzero, not of a market.')

pair        pre-tax (t)      + tax term (sd)     = total (t)      mean%   var%
MUB-AGG    +1.73 (+0.31)   +10.21 (sd 5.72)   +11.94 (+2.22)    85.5   0.30
VTEB-AGG   +4.34 (+0.65)    +9.47 (sd 4.74)   +13.81 (+2.10)    68.6   0.37
SUB-BIL    +2.49 (+0.69)    +4.22 (sd 6.87)    +6.71 (+2.02)    62.9   1.87
HYD-LQD    +6.77 (+0.57)   +12.83 (sd 6.01)   +19.60 (+1.67)    65.5   0.09
MUB-VCIT  -10.60 (-1.50)   +12.35 (sd 6.32)    +1.75 (+0.25)      --   0.32

Every row crossing |t| = 2 does so on a term carrying < 2% of the variance.
That is a test of whether the tax code is nonzero, not of a market.


## The bracket ladder — MUB vs VCIT (like-for-like) and MUB vs AGG (not)

In [5]:
print('rate   MUB-VCIT bps (t)      MUB-AGG bps (t)')
for r, v, vt, g, gt in zip(R['ladder_rates'], R['ladder_vcit'], R['ladder_vcit_t'],
                           R['ladder_agg'], R['ladder_agg_t']):
    print('%5.1f%%  %+7.2f (%+5.2f)      %+7.2f (%+5.2f)' % (r, v, vt, g, gt))
print()
print('MUB-VCIT at 40.8%%: %+.2f bps/mo (%+.2f pp/yr)  HAC t=%+.2f  boot CI [%+.1f, %+.1f]'
      % (R['d_mub_vcit'], R['ann_mub_vcit'], R['t_mub_vcit'],
         R['ci_mub_vcit_lo'], R['ci_mub_vcit_hi']))
print('MUB-AGG  at 40.8%%: %+.2f bps/mo (%+.2f pp/yr)  HAC t=%+.2f  boot CI [%+.1f, %+.1f]'
      % (R['d_mub_agg'], R['ann_mub_agg'], R['t_mub_agg'],
         R['ci_mub_agg_lo'], R['ci_mub_agg_hi']))
print('excess-of-after-tax-cash Sharpe, MUB vs VCIT: %+.2f vs %+.2f'
      % (R['sh_mub_vcit_m'], R['sh_mub_vcit_t']))

rate   MUB-VCIT bps (t)      MUB-AGG bps (t)
  0.0%   -10.63 (-1.50)        +1.70 (+0.31)
 24.0%    -3.37 (-0.48)        +7.71 (+1.42)
 27.8%    -2.22 (-0.32)        +8.66 (+1.60)
 35.8%    +0.20 (+0.03)       +10.66 (+1.98)
 40.8%    +1.72 (+0.25)       +11.91 (+2.22)

MUB-VCIT at 40.8%: +1.72 bps/mo (+0.21 pp/yr)  HAC t=+0.25  boot CI [-12.5, +15.6]
MUB-AGG  at 40.8%: +11.91 bps/mo (+1.43 pp/yr)  HAC t=+2.22  boot CI [+0.9, +22.5]
excess-of-after-tax-cash Sharpe, MUB vs VCIT: +0.47 vs +0.35


The MUB-AGG ladder is the mechanism above in motion: *t* climbs monotonically from +0.31 to +2.22 purely because the bracket multiplies a near-constant coupon stream. The pairings that clear |*t*| ≥ 2 on the pooled sample — MUB/AGG (+2.22), VTEB/AGG (+2.09), SUB/BIL (+2.01) — all do so on that constant. Reading the bottom row of a bracket ladder as a significance test is the specific mistake this study makes easy.

## Era cut (split 2017-01) — nothing survives in halves

In [6]:
print('MUB-VCIT  pre-2017 (n=%3d): %+6.2f bps  t=%+5.2f' % (R['era_vcit_e_n'], R['era_vcit_e'], R['era_vcit_e_t']))
print('MUB-VCIT  2017-on  (n=%3d): %+6.2f bps  t=%+5.2f   <- sign flips' % (R['era_vcit_l_n'], R['era_vcit_l'], R['era_vcit_l_t']))
print('MUB-AGG   pre-2017 (n=%3d): %+6.2f bps  t=%+5.2f' % (R['era_agg_e_n'], R['era_agg_e'], R['era_agg_e_t']))
print('MUB-AGG   2017-on  (n=%3d): %+6.2f bps  t=%+5.2f   <- neither half clears 2' % (R['era_agg_l_n'], R['era_agg_l'], R['era_agg_l_t']))

MUB-VCIT  pre-2017 (n= 85):  -2.40 bps  t=-0.23
MUB-VCIT  2017-on  (n=114):  +4.73 bps  t=+0.51   <- sign flips
MUB-AGG   pre-2017 (n=111):  +9.96 bps  t=+1.25
MUB-AGG   2017-on  (n=114): +13.76 bps  t=+1.92   <- neither half clears 2


## The PROXY sweeps — state, capital gains, income floor

In [7]:
print('state 0%%             : %+5.2f bps/mo' % R['state_none'])
print('state 13.3%%, national: %+5.2f bps/mo   (income is mostly out-of-state -> taxed anyway)' % R['state_ca_out'])
print('state 13.3%%, in-state: %+5.2f bps/mo (t=%+.2f)  <- best case, still nowhere near 2' % (R['state_ca_in'], R['state_ca_in_t']))
print()
print('cap-gains  0%% (unrealised, default): %+5.2f bps/mo' % R['cg_0'])
print('cap-gains 23.8%% (20%% + NIIT)       : %+5.2f bps/mo' % R['cg_238'])
print()
print('income floor on : %+6.3f bps/mo' % R['floor_on'])
print('income floor off: %+6.3f bps/mo   <- the reconstruction choice is immaterial' % R['floor_off'])

state 0%             : +1.72 bps/mo
state 13.3%, national: +2.47 bps/mo   (income is mostly out-of-state -> taxed anyway)
state 13.3%, in-state: +4.53 bps/mo (t=+0.65)  <- best case, still nowhere near 2

cap-gains  0% (unrealised, default): +1.72 bps/mo
cap-gains 23.8% (20% + NIIT)       : +2.32 bps/mo

income floor on : +1.717 bps/mo
income floor off: +1.716 bps/mo   <- the reconstruction choice is immaterial


## Costs (irrelevant) and borrow (fatal)

As an asset-location choice the friction is a single round trip out of the incumbent, amortised. As a long-short spread the short taxable leg pays borrow — and that is what kills it.

In [8]:
print('cost  0 bps one-way: %+5.2f bps/mo' % R['cost0'])
print('cost 25 bps one-way: %+5.2f bps/mo   <- one trade in 17 years' % R['cost25'])
print()
print('borrow   0 bps/yr: MUB-VCIT %+6.2f   MUB-AGG %+6.2f' % (R['borrow0_vcit'], R['borrow0_agg']))
print('borrow  25 bps/yr: MUB-VCIT %+6.2f   MUB-AGG %+6.2f (t=%+.2f)  <- both broken'
      % (R['borrow25_vcit'], R['borrow25_agg'], R['borrow25_agg_t']))
print('borrow 100 bps/yr: MUB-VCIT %+6.2f   MUB-AGG %+6.2f' % (R['borrow100_vcit'], R['borrow100_agg']))

cost  0 bps one-way: +1.75 bps/mo
cost 25 bps one-way: +1.50 bps/mo   <- one trade in 17 years

borrow   0 bps/yr: MUB-VCIT  +1.72   MUB-AGG +11.91
borrow  25 bps/yr: MUB-VCIT  -0.37   MUB-AGG  +9.83 (t=+1.83)  <- both broken
borrow 100 bps/yr: MUB-VCIT  -6.62   MUB-AGG  +3.58


> 💡 **In plain words.** Choosing which bond fund to own costs one trade, so friction is a rounding error. Trying to *arbitrage* the pair means shorting the taxable fund, and the borrow fee on a corporate-bond ETF is larger than the entire edge.

## The one arm with a signal — and therefore exactly one execution lag

Trailing 12-month after-tax income yields known at the close of month *t* decide the leg held in month *t+1*. 3 bps one-way × NAV per switch.

In [9]:
print('MUB/VCIT overlay: n=%d  in-muni %.1f%%  switches %d over ~17 years'
      % (R['ov_n'], R['ov_in'], R['ov_switches']))
print('  overlay %+.2f%%/yr vs simply holding the muni %+.2f%%/yr'
      % (R['ov_ann'], R['ov_hold']))
print('  edge %+.2f bps/mo (HAC t=%+.2f) -> the bracket question has no timing dimension'
      % (R['ov_edge'], R['ov_t']))

MUB/VCIT overlay: n=199  in-muni 93.0%  switches 3 over ~17 years
  overlay +3.33%/yr vs simply holding the muni +3.05%/yr
  edge +2.31 bps/mo (HAC t=+0.90) -> the bracket question has no timing dimension


## Live synthetic control — machinery proof, never a stamp

Forty years of monthly data, eight seeds. **Planted world:** the taxable leg yields 150 bp more pre-tax, so the true break-even is exactly 33.33% and the after-tax difference must flip sign across it. **Twin null:** identical yields and duration, so the break-even must collapse to ~0 and no *pre-tax* difference may be manufactured.

Note what the null deliberately does *not* claim: on twins the *after-tax* difference at a positive bracket is large and significant, because it is pure arithmetic (`tau × yield`). That is the effect being measured; the control's job is to show the machinery attributes none of it to a market edge.

In [10]:
import os, sys
sys.path.insert(0, os.path.abspath('..'))
sys.path.insert(0, os.path.abspath(os.path.join('..','..','..')))
import numpy as np
from after_tax import data, strategy as st
for ss, label in [(1.0, 'planted 150bp gap'), (0.0, 'twin null      ')]:
    errs = []
    for s in range(8):
        panel, truth = data.synthetic_panel(signal_strength=ss, seed=952 + s)
        errs.append(st.breakeven_rate(panel, 'muni', 'taxable')['breakeven']
                    - truth['planted_breakeven'])
    panel, truth = data.synthetic_panel(signal_strength=ss, seed=952)
    d = st.synthetic_detect(panel)
    errs = np.array(errs)
    print('%s true %6.2f%%  recovered %6.2f%%  (8 seeds: mean err %+.2f pp, max |err| %.2f pp)'
          % (label, truth['planted_breakeven']*100, d['breakeven']*100,
             errs.mean()*100, np.abs(errs).max()*100))
    print('    pre-tax diff %+6.2f bps (t=%+5.2f)   below break-even %+6.2f bps -> above %+6.2f bps'
          % (d['pretax_diff_bps'], d['pretax_t'],
             d['diff_below_breakeven'], d['diff_above_breakeven']))

planted 150bp gap true  33.33%  recovered  38.97%  (8 seeds: mean err +3.82 pp, max |err| 10.33 pp)
    pre-tax diff -14.61 bps (t=-8.29)   below break-even  -5.63 bps -> above  +5.62 bps


twin null       true   0.00%  recovered   8.05%  (8 seeds: mean err +3.69 pp, max |err| 9.83 pp)
    pre-tax diff  -2.52 bps (t=-1.46)   below break-even  -2.52 bps -> above  +4.69 bps


### The t-stat demonstration, live on the twin null

Two **statistically identical** bonds, nothing planted. Watch the after-tax HAC *t* walk past 2 as the assumed bracket rises. Nothing arrives but arithmetic — which is why no |*t*| ≥ 2 in the real-tape tables above can be read as a market effect.

In [11]:
panel, _ = data.synthetic_panel(signal_strength=0.0, seed=952)
print('twin null: identical yields, identical duration, zero planted edge')
for f in (0.0, 0.10, 0.20, 0.30, 0.40):
    r = st.race(panel, 'muni', 'taxable', st.tax_profile(f, 0.0, 0.0),
                cash='cash', cost_bps=0.0)
    print('  bracket %4.0f%%   after-tax diff %+7.2f bps/mo   HAC t = %+5.2f'
          % (100*f, r['diff_bps'], r['t_diff']))
dec = st.tax_constant_decomposition(panel, 'muni', 'taxable', st.tax_profile(0.37))
print('\nat 40.8%%: tax term is %.1f%% of the mean and %.2f%% of the variance'
      % (100*dec['mean_share_tax'], 100*dec['var_share_tax']))

twin null: identical yields, identical duration, zero planted edge


  bracket    0%   after-tax diff   -2.52 bps/mo   HAC t = -1.46
  bracket   10%   after-tax diff   +0.61 bps/mo   HAC t = +0.36


  bracket   20%   after-tax diff   +3.73 bps/mo   HAC t = +2.17
  bracket   30%   after-tax diff   +6.86 bps/mo   HAC t = +3.99


  bracket   40%   after-tax diff   +9.98 bps/mo   HAC t = +5.81

at 40.8%: tax term is 124.6% of the mean and 0.00% of the variance


## Verdict

- **Signal — Weak.** Exactly one estimand here is identified, and it is not an outperformance. The **income-leg break-even** is tight — **26.7%** [23.4, 29.7] vs VCIT, **28.9%** [27.1, 30.7] vs LQD — measured on realised distributions net of fees, and it sits below the top two brackets. Everything built on top of it fails. The **total-return break-even is not identified**: 35.0% point, 95% CI [-10.9%, 81.5%], 41% of draws above the top US rate — its numerator never clears |*t*| ≥ 2 in any pairing (largest -1.50). The **after-tax race is a dead heat**: +1.72 bps/mo (+0.21 pp/yr), HAC *t* = +0.25, CI [-12.5, +15.6]. And the three pairings clearing |*t*| ≥ 2 clear it on a **tax constant** worth 86% of the mean and 0.30% of the variance — the same lift appears on the synthetic twin null with nothing planted. **No |*t*| ≥ 2 in this study is evidence of a market effect**, and none survives either era half. The control recovers a planted 33.33% break-even to within 10.3 pp across 8 seeds and stays quiet on twins (pre-tax *t* = -1.46), so the dead heat is the muni market, not the harness.
- **Tradability — Fragile.** Real and free as an *allocation* (one round trip, 0.03 bps/mo amortised; in a top bracket hold munis in the taxable account, which the tight income-leg break-even does support), but the total-return margin sits inside the noise, the answer rides on an assumed bracket, the crossover cannot be pinned better than [-11%, 82%], and as a *trade* -0.37 bps/mo at 25 bps/yr borrow is the whole story.